### PR TITLE
xc: strips nl from str

### PR DESCRIPTION
**Pull request by msnmz**
Removes CRLF from the input string

### CODE REVIEW

**Comment by mouchtaris** [comment]
How about we make it actually split the input string at new lines and output the NULL-sep list?

File: component/cargo/xc/src/lib/lib.rs [hunk 1]
```diff
@@ -52,6 +54,27 @@ pub fn app<I: io::Read, O: io::Write>(
...
+                write!(
+                    stdout,
+                    "{}",
+                    buffer.trim_end_matches('\n').trim_end_matches('\r')
+                )?;
+                Ok(())
+            }
+            NULL_SEPARATED => {
+                let lines: Vec<_> =
+                    buffer.lines().filter(|line| !line.is_empty()).collect();
```
Comment (mouchtaris) [inline component/cargo/xc/src/lib/lib.rs]: I don't think we need the `empty` check, so the whole block can be simplified to one iterator chain after `buffer.lines()`.

File: component/cargo/xc/src/lib/lib.rs [hunk 2]
```diff
@@ -52,6 +54,27 @@ pub fn app<I: io::Read, O: io::Write>(
...
+                let last_index = lines.len() - 1;
```
Comment (mouchtaris) [inline component/cargo/xc/src/lib/lib.rs]: A better trick is

```rust
let mut separator = "";
for item in iter {
  write!(stdout, "{}{}", separator, item);
  separator = "\x00";
}
```

File: component/cargo/xc/src/lib/lib.rs [hunk 3]
```diff
@@ -52,6 +54,27 @@ pub fn app<I: io::Read, O: io::Write>(
 
                 Ok(())
             }
+            NL_TRIMMED => {
```
Comment (mouchtaris) [inline component/cargo/xc/src/lib/lib.rs]: I think because `NL_TRIMMED` has the same semantics as an one-line `NULL_SEPARATED`, we can remove it.

**Review by mouchtaris — CHANGES_REQUESTED** [review]
Feedback:

File: component/cargo/xc/src/lib/lib.rs (hunk 2 above)
Comment (msnmz) [reply component/cargo/xc/src/lib/lib.rs]: Loved that trick 👍

File: component/cargo/xc/src/lib/lib.rs [hunk 4]
```diff
@@ -52,6 +53,14 @@ pub fn app<I: io::Read, O: io::Write>(
 
                 Ok(())
             }
+            NULL_SEPARATED => {
+                let mut separator = "";
+                buffer.lines().for_each(move |line| {
+                    let _ = write!(out, "{}{}", separator, line);
+                    separator = "\x00";
```
Comment (mouchtaris) [inline component/cargo/xc/src/lib/lib.rs]: I think this assignment can be [avoided](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold) as well.

**Review by mouchtaris — CHANGES_REQUESTED** [review]

**Pull request merged**